Sdh okta594449 cors (#4064)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/oie-upgrade-plan-embedded-upgrades/main/index.md

<ApiLifecycle access="ie" />

We understand that upgrades can be stressful. We've carefully considered how you can break up the upgrade steps to ensure that you maintain the best user experience across your applications.

> We don’t recommend doing this upgrade all at once, but rather in sections with breaks for testing. This process is designed to be non-disruptive and iterative over a time. Make the upgrade steps part of your normal product development process.

Author: that was a typo - needs to just be "iterative over time".

    We recommend the following order, but prioritize the order of your upgrade workflows to suit your needs.

    * **Self-Service Password Recovery:** Replace with the remediation form of the password reset flow. This may also mean removing features like security questions and some other configurations from the Password Authenticator's settings. See [Upgrade your application to the Identity Engine SDK](/docs/guides/oie-upgrade-api-sdk-to-oie-sdk/nodejs/main/) for detailed steps by language. <!-- (/docs/guides/oie-upgrade-api-sdk-to-oie-sdk/-/main/) -->

    * **Self-Service Registration:** The Identity Engine SDK provides full support for user sign-up flows. See the [Self-service registration integration guide](/docs/guides/oie-embedded-sdk-use-case-self-reg/-/main/) for more information on how to integrate the sign up use case into your application using the Identity Engine SDK.

    * **Authentication**

>         * **Sign In:** In your new Identity Engine org, the Organization Sign-On Policy and app sign-on rules are mapped to a global session policy and [corresponding authentication policy](https://help.okta.com/okta_help.htm?type=oie&id=ext-about-asop) respectively. Test and make sure that your user experience still works the same as what was configured in Classic Engine. If you have applications that employ multifactor authentication, make sure that they still authenticate in the same way. Applications should still work after the policies are upgraded.

Reviewer: Should the bolded element be **Sign-On** to match the resy of the element?

    * Have logic in your code to send some users into the Classic Engine workflow and some to Identity Engine.
    * Load balance network traffic to different application instances, containers, or web servers to send some to a Classic Engine version of your app or to an Identity Engine version.
    * Increase this over time, as you see fit.
    * Remove code or shutdown alternate versions when everything is functional.

## Troubleshoot

This section details common issues and errors that you may encounter during your upgrade, and likely causes.

| Issue / error | Cause |
| ------ | -------------------------- |

> | Error message: "The requested feature isn't available in this environment" | This is reported when your [Okta org](/docs/concepts/okta-organizations/) hasn't been upgraded to Identity Engine and you try to turn on [interaction code](/docs/concepts/interaction-code/) in the embedded SDK/Sign-In Widget. See [Identify your Okta solution](https://help.okta.com/okta_help.htm?type=oie&id=ext-oie-version) to determine your Okta version. |

Reviewer: Maybe something like:
```suggestion
| Error message: "The requested feature isn't available in this environment" | This is reported when you attempt to enable [interaction code](/docs/concepts/interaction-code/) in the embedded SDK/Sign-In Widget on a non-OIE [Okta org](/docs/concepts/okta-organizations/). See [Identify your Okta solution](https://help.okta.com/okta_help.htm?type=oie&id=ext-oie-version) to determine your Okta version. |
```

Author: I'm going to leave this one as is - we aren't supposed to use OIE in outward facing docs. It's an internal term.

> | &nbsp; | This error is also reported when you've upgraded your org to Identity Engine and interaction code is turned on in the embedded SDK, but it isn't turned on in your app or authorization server. |

Reviewer: Something like:
```suggestion
| &nbsp; | This is also reported when you've enabled interaction code in the embedded SDK, but not in your app or authorization server. |
```

Author: since this is an upgrade doc, I'd like to keep the reference to the steps they took to get there, which includes first upgrading their org to OIE.

> | &nbsp; | This error is also reported when you've upgraded your org to Identity Engine and interaction code is turned on in the embedded SDK, but it isn't turned on in your app or authorization server. |

Author: ```suggestion
| &nbsp; | This is also reported when you've upgraded your org to Identity Engine and enabled interaction code in the embedded SDK, but it isn't enabled in your app or authorization server. |
```
